src/Algebra/Graph/Acyclic/Labelled/AdjacencyMap/Algorithm.hs

```diff
@@ -0,0 +1,47 @@
+module Algebra.Graph.Acyclic.Labelled.AdjacencyMap.Algorithm where
+
+import Algebra.Graph.Acyclic.Labelled.AdjacencyMap
+import qualified Algebra.Graph.AdjacencyMap.Algorithm as AM
+import Algebra.Graph.Label
+import qualified Algebra.Graph.Labelled.AdjacencyMap as LAM
+import Data.Map.Strict (Map, (!))
+import qualified Data.Map.Strict as Map
+import Data.Maybe (fromMaybe)
+import Data.List (foldl')
+
+-- TODO: Replace this function with 'skeleton' from Acyclic.Labelled to remove the use of fromMaybe
+-- TODO: Make 'topSort' more efficient
+-- TODO: Add examples and tests for 'topSort'
+topSort :: (Ord a) => AdjacencyMap e a -> [a]
+topSort = fromMaybe [] . AM.topSort . LAM.skeleton . fromAcyclic
+
+-- TODO: Improve documentation for 'fold'
+-- TODO: Add examples and tests for 'fold'
+-- | Compute the final state by using the initial state and
+-- traversing the entire graph in topological order.
+fold :: (Ord a) => (e -> a -> a -> s -> s) -> s ->  AdjacencyMap e a -> s
```

Comment:
We need to explain what the first argument does, clarifying the order of arguments. Also, we need a few simple examples.